### DISCUSSION TITLE
Special configuration with 3 meters

### DISCUSSION THREAD

**Discussion by AlexJacu** [hif2k1/battery_sim#115] [Q&A]
We have the following configuration:
1. a photovoltaic system with its own production meter. This shows positive values for the amount of electricity fed in/produced and fed into the grid.
2. a house connection with a meter that displays the consumption with positive values. (It can also display the feed-in, but we only feed in with the production meter)
3. a virtual meter that displays the current consumption in positive values and feed-in as negative values. But the values are only correct as W and not as kWh
Unfortunately, the virtual meter does NOT show consumption and feed-in separately, but in the same entity.

How can we map this in order to find a suitable simulation for a battery?

Wir haben folgende Konfiguration:
1. Eine Photovaltaikanlage mit einem eigenen Produktionszähler. Dieser zeigt positive Werte für die Eingespeiste/Produzierte Strommenge an, die in das Netz eingespeist wird.
2. Einen Hausanschluss mit einem Zähler, der den Verbrauch mit positiven Werten anzeigt. (Er kann auch die Einspeisung anzeigen, aber wir speisen nur mit dem Produktionszähler ein)
3. Einen virtuellen Zähler, der den aktuellen Verbrauch in positive und Einspeisung als negative werte anzeigt. Aber die Werte sind nur als W und nicht als kWh richtig
Der virtuelle Zähler zeigt leider Verbrauch und Einspeisung NICHT getrennt an, sondern in der gleichen Entität.

Wie können wir dies abbilden, um eine passende Simulation für einen Akku zu finden?

**Comment by hif2k1** [comment]
Hmm, that is tricky one. So essentially your solar is totally separate to your house and they both connect to the grid independently. I think you're going to have to create some template sensors to sit between the true meters and the battery. 
It's probably best to start with the virtual meter and then create a helper or template sensor by adding something like this to your configuration yaml:

```
template:
  - sensor:
      - name: "Import W"
        unit_of_measurement: "W"
        state: >
          {% set virtual_meter = state_attr('sensor.virtual_meter') | float %}
          {% if virtual_meter > 0} 
              {{ virtual_meter/1000 }}
          {% else %} 0.0 {% endif %}
      - name: "Export W"
        unit_of_measurement: "W"
        state: >
          {% set virtual_meter = state_attr('sensor.virtual_meter') | float %}
          {% if virtual_meter < 0} 
               {{ virtual_meter/-1000 }}
          {% else %} 0.0 {% endif %}
```
This will give you separate import and export meters in kilowatts. You would then need to integrate these to get the reading in kWh. Luckily there is a built in integration helper in home assistant under settings -> Devices and Services. Again make two new integration helpers each taking one of the template sensors you just made as an input. The outputs of these should show the import and export overall for your system and can be used as inputs for the battery.

Having said all that I'm not sure how a real battery would handle this situation so that's something you might need to look into.
